solver/llbsolver/file/backend.go

```diff
@@ -250,7 +211,16 @@ func docopy(ctx context.Context, src, dest string, action pb.FileActionCopy, u *
 	return nil
 }
 
+// NewFileOpBackend returns a new file operation backend. The executor is currently only used for Windows,
+// and it is used to construct the readUserFn field set in the returned Backend.
+func NewFileOpBackend(exec executor.Executor) *Backend {
```

Comment:
I'd like to avoid passing this `Executor` in here (and import on buildkit/executor). And instead pass a function/interface only for readuser (ideally only on windows, in linux you can just pass nil).

Reply:
This will mean having code that reads the user defined in more than one place (this package and somewhere else). Also, if we only pass in that function/interface for one OS, it creates a special case that will add cognitive burden for future readers of this code (one more thing to keep track of). I can do that if you wish, but it might be better to pass in that function/interface for all OSs, just to keep things consistent and in one place.

If we decide to go that route, we can define a new interface in `solver/llbsolver/file`, implement it in `solver/llbsolver/ops` package (the Windows version will call into the executor which is already imported in `exec.go`, and pass that implementation into Backend{}`.

How does that sound?

Reply:
>  I can do that if you wish, but it might be better to pass in that function/interface for all OSs, just to keep things consistent and in one place.

Fine by me if you prefer that. I thought about it from the mindset that file can do its job without special configuration but if user wants to pass in custom readUser they can do this. This doesn't necessarily need to be just OS-based in the future but using `/etc/passwd` from some other snapshot or some other user database.

> we can define a new interface 

If it is just one function it can just be a callback as well.

Reply:
I moved the `readUser()` code and defined a callback type. Have a look and let me know if you would prefer it done differently.